overlord/snapstate/check_snap.go

```diff
@@ -369,8 +369,38 @@ func checkBases(st *state.State, snapInfo, curInfo *snap.Info, flags Flags) erro
 	return fmt.Errorf("cannot find required base %q", snapInfo.Base)
 }
 
+func checkEpochs(_ *state.State, snapInfo, curInfo *snap.Info, _ Flags) error {
+	if curInfo == nil {
+		return nil
+	}
+	if snapInfo.Epoch.CanRead(curInfo.Epoch) {
+		return nil
+	}
+	return fmt.Errorf("cannot refresh snap %q as new epoch (%s) can't read old epoch (%s)", snapInfo.InstanceName(), snapInfo.Epoch, curInfo.Epoch)
```

Comment:
as new revision %d with epoch (%s) can't read current epoch (%s) ?

Reply:
Hmm.

`"cannot refresh snap %q as new revision has epoch %s that can't read current revision's epoch (.%s)"`?

Could also add the new revision in parentheses if it isn't local (if local, it won't be useful)

Reply:
sounds good (also the revision in parentheses bit)